addon/components/frost-file-picker.js

```diff
@@ -75,10 +76,22 @@ export default Component.extend({
     this.set('isDragging', state)
   },
   dragOver (event) {
-    this._setDragging(event, true)
+    if (this.get('validateDrag')) {
+      if (this.validateDrag(event)) {
+        this._setDragging(event, true)
+      }
+    } else {
+      this._setDragging(event, true)
+    }
```

Comment:
A couple issues with this:
1) I don't think it's a good idea to mix using `this.get('validateDrag')` and `this.validateDrag` I'd stick with just the latter
2) I don't think all the nested blocks are necessary, I'd probably refactor it to this: 
```js
if (this.validateDrag && !this.validateDrag(event)) {
  return
}

this._setDragging(event, true)
```